src/components/TodoApp.tsx

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { createTodo, getTodos } from '../api/todos';
+import { Todo } from '../types/Todo';
+import { Menu } from './Menu';
+import { TodoList } from './TodoList';
+
+export const TodoApp: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    getTodos().then(setTodos);
+  }, []);
+
+  const findMaxId = () => {
+    return Math.max(...todos.map(todo => todo.id));
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.length) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      title,
+      id: findMaxId() + 1,
+      userId: 30,
+      completed: false,
+    };
+
+    await createTodo(newTodo);
+
+    getTodos().then(setTodos);
+
+    setTitle('');
+  };
+
+  const itemsLeft = todos.filter(todo => !todo.completed).length;
+
+  let displayedTodos;
+
+  switch (pathname) {
+    case '/active':
+      displayedTodos = todos.filter(todo => !todo.completed);
+      break;
+
+    case '/completed':
+      displayedTodos = todos.filter(todo => todo.completed);
+      break;
+
+    default:
+      displayedTodos = todos;
+  }
```

Comment:
GJ 🔥 
It's working as it shoud! I'll just share some of my thougths.
We use default case for covering some unexpected cases. If you have specific result for specific case then you have to announce it.